GUI/ChooseKSPInstance.cs

```diff
@@ -76,55 +84,63 @@ private void SelectButton_Click(object sender, EventArgs e)
+            var instName = selected?.Tag as string;
+            if (instName == null)
+            {
+                return;
+            }
 
             if (SetAsDefaultCheckbox.Checked)
             {
-                manager.SetAutoStart(instance);
+                _manager.SetAutoStart(instName);
             }
 
-            manager.SetCurrentInstance(instance);
+            _manager.SetCurrentInstance(instName);
             DialogResult = DialogResult.OK;
             Close();
         }
 
         private void KSPInstancesListView_SelectedIndexChanged(object sender, EventArgs e)
         {
-            var has_instance = KSPInstancesListView.SelectedItems.Count != 0;
-            SetButtonsEnabled(has_instance);
+            UpdateButtonState();
         }
 
         private void KSPInstancesListView_DoubleClick(object sender, EventArgs r)
         {
-            var has_instance = KSPInstancesListView.SelectedItems.Count != 0;
-            if(has_instance)
-                UseSelectedInstance();
+            if (HasSelections) UseSelectedInstance();
         }
 
         private void RenameButton_Click(object sender, EventArgs e)
         {
-            var instance = (string) KSPInstancesListView.SelectedItems[0].Tag;
+            var instance = (string)KSPInstancesListView.SelectedItems[0].Tag;
 
-            renameInstanceDialog = new RenameInstanceDialog();
-            if (renameInstanceDialog.ShowRenameInstanceDialog(instance) == DialogResult.OK)
-            {
-                manager.RenameInstance(instance, renameInstanceDialog.GetResult());
-                UpdateInstancesList();
-            }
+            // show the dialog, and only continue if the user selected "OK"
+            _renameInstanceDialog = new RenameInstanceDialog();
+            if (_renameInstanceDialog.ShowRenameInstanceDialog(instance) != DialogResult.OK) return;
+
+            // proceed with instance rename
+            _manager.RenameInstance(instance, _renameInstanceDialog.GetResult());
+            UpdateInstancesList();
         }
 
         private void Forget_Click(object sender, EventArgs e)
         {
-            var instance = (string)KSPInstancesListView.SelectedItems[0].Tag;
-            manager.RemoveInstance(instance);
-            UpdateInstancesList();
-
+            foreach (var instance in KSPInstancesListView.SelectedItems.OfType<ListViewItem>().Select(item => item.Tag as string))
+            {
+                _manager.RemoveInstance(instance);
+                UpdateInstancesList();
+            }
         }
 
-        private void SetButtonsEnabled(bool has_instance)
+        private void UpdateButtonState()
         {
-            ForgetButton.Enabled = RenameButton.Enabled = SelectButton.Enabled = SetAsDefaultCheckbox.Enabled = has_instance;
+            ForgetButton.Enabled = RenameButton.Enabled = SelectButton.Enabled = SetAsDefaultCheckbox.Enabled = _manager.Instances.Count > 0;
```

Comment:
If you use hasSelections here, the buttons are greyed out correctly, and I don't get any exceptions

`ForgetButton.Enabled = RenameButton.Enabled = SelectButton.Enabled = SetAsDefaultCheckbox.Enabled = HasSelections;`